pkg/volume/aws_ebs/aws_ebs.go

```diff
@@ -189,10 +189,26 @@ func getVolumeSource(
 func (plugin *awsElasticBlockStorePlugin) ConstructVolumeSpec(volName, mountPath string) (*volume.Spec, error) {
 	mounter := plugin.host.GetMounter()
 	pluginDir := plugin.host.GetPluginDir(plugin.GetPluginName())
-	sourceName, err := mounter.GetDeviceNameFromMount(mountPath, pluginDir)
+	volumeID, err := mounter.GetDeviceNameFromMount(mountPath, pluginDir)
 	if err != nil {
 		return nil, err
 	}
+	// This is a workaround to fix the issue in converting aws volume id from globalPDPath
+	// There are three aws volume id formats and their globalPDPaths are:
+	// aws:///vol-1234 (../aws-ebs/mounts/aws/vol-1234)
+	// aws://us-east-1/vol-1234 (../aws-ebs/mounts/aws/us-east-1/vol-1234)
+	// vol-1234 (../aws-ebs/mounts/vol-1234)
+	// This code is for dealing with the first two cases.
+	sourceName := volumeID
+	if strings.HasPrefix(volumeID, "aws/") {
+		if len(strings.Split(volumeID, "/")) > 2 {
+			sourceName = strings.Replace(volumeID, "aws/", "aws://", 1)
+		} else {
+			sourceName = strings.Replace(volumeID, "aws/", "aws:///", 1)
+		}
+		glog.V(4).Info("Convert aws volume name from %q to %q ", volumeID, sourceName)
+	}
```

Comment:
This looks right.

If I'm going to nit-pick, it might be safer to split the string, and then glog.Warning if it doesn't match one of the forms we expect.

i.e. we expect either aws/vol-12345678 or aws/us-east-1b/vol-12345678, but we should log if it doesn't look like that.

Bonus: code might be easier to understand also!